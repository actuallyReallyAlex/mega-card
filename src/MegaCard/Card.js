import React from 'react'
import { Box, Text } from 'grommet'
import { Info } from 'grommet-icons'
import { buildShadow, colors, defaultProps, propTypes } from '../constants'

const Card = ({ background, color, data, shadowSize, shadowType, title }) => {
  return (
    <Box
      align="center"
      background={background}
      direction="row"
      gap="medium"
      justify="between"
      margin="small"
      pad="medium"
      round="medium"
      style={{ boxShadow: buildShadow(shadowSize, shadowType) }}
    >
      <Box>
        <Box
          height="8px"
          round="full"
          style={{ backgroundColor: colors[color] }}
          width="8px"
        />
      </Box>
      <Box margin={{ right: 'medium' }}>
        <Text size="16px" weight="bold">
          {title}
        </Text>
        <Text size="small" color="dark-4">
          {data}
        </Text>
      </Box>
      <Box>
        <Box
          align="center"
          height="35px"
          justify="center"
          round="full"
          style={{ backgroundColor: colors[color] }}
          width="35px"
        >
          <Info color="white" size="20px" />
        </Box>
      </Box>
    </Box>
  )
}

Card.propTypes = propTypes

Card.defaultProps = defaultProps

export default Card
