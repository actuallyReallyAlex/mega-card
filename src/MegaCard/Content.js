import React from 'react'
import { Box, Heading, Text } from 'grommet'
import { colors, propTypes } from '../constants'

const Content = ({ data, title, titleColor }) => {
  return (
    <Box align="start" margin={{ right: 'medium' }}>
      <Text size="16px" style={{ color: colors[titleColor] }} weight="bold">
        {title}
      </Text>
      <Heading
        color="dark-2"
        level="5"
        margin={{ bottom: 'none', top: 'xsmall' }}
        style={{ fontSize: '1.25rem' }}
      >
        {data}
      </Heading>
    </Box>
  )
}

Content.propTypes = {
  data: propTypes.data,
  title: propTypes.title,
  titleColor: propTypes.titleColor
}

export default Content
