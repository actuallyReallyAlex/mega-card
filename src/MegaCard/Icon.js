import React from 'react'
import { Box } from 'grommet'
import { colors, icons, propTypes } from '../constants'

const Icon = ({ color, icon }) => {
  const Figure = icons[icon]
  return (
    <Box>
      <Box
        align="center"
        height="42px"
        justify="center"
        round="full"
        style={{ backgroundColor: colors[color] }}
        width="42px"
      >
        <Figure color="white" size="28px" />
      </Box>
    </Box>
  )
}

Icon.propTypes = {
  color: propTypes.color.isRequired,
  icon: propTypes.icon.isRequired
}

export default Icon
