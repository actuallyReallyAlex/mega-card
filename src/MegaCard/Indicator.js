import React from 'react'
import { Box } from 'grommet'
import { colors, propTypes } from '../constants'

const Indicator = ({ color }) => {
  return (
    <Box>
      <Box
        height="8px"
        round="full"
        style={{ backgroundColor: colors[color] }}
        width="8px"
      />
    </Box>
  )
}

Indicator.propTypes = {
  color: propTypes.color
}

export default Indicator
