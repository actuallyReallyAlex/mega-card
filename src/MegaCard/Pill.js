import React from 'react'
import { Box, Text } from 'grommet'
import { Info } from 'grommet-icons'
import { buildShadow, colors, defaultProps, propTypes } from '../constants'

const Pill = ({
  background,
  color,
  data,
  indicator,
  shadowSize,
  shadowType,
  title
}) => {
  return (
    <Box
      align="center"
      background={background}
      direction="row"
      gap="medium"
      justify="between"
      margin="small"
      pad="medium"
      round="large"
      style={{ boxShadow: buildShadow(shadowSize, shadowType) }}
    >
      {indicator && (
        <Box>
          <Box
            style={{ backgroundColor: colors[color] }}
            height="8px"
            round="full"
            width="8px"
          />
        </Box>
      )}
      <Box margin={{ right: 'medium' }}>
        <Text size="16px" weight="bold">
          {title}
        </Text>
        <Text size="small" color="dark-4">
          {data}
        </Text>
      </Box>
      <Box>
        <Box
          align="center"
          height="35px"
          justify="center"
          round="full"
          style={{ backgroundColor: colors[color] }}
          width="35px"
        >
          <Info color="white" size="20px" />
        </Box>
      </Box>
    </Box>
  )
}

Pill.propTypes = propTypes

Pill.defaultProps = defaultProps

export default Pill
