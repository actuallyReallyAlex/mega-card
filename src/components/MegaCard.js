import React from 'react'
import PropTypes from 'prop-types'

const MegaCard = ({ version }) => {
  return (
    <div>
      <span>MegaCard {version}</span>
    </div>
  )
}

MegaCard.propTypes = {
  version: PropTypes.string
}

MegaCard.defaultProps = {
  version: '1.0'
}

export default MegaCard
