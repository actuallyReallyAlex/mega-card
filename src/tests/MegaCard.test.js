import React from 'react'
import MegaCard from '../index'
import { shallow } from 'enzyme'

const defaultProps = {
  data: 'Test Data',
  title: 'Title'
}

describe('<MegaCard />', () => {
  afterEach(() => {
    // defaultProps.handler.mockClear()
  })

  test('Should render the <MegaCard /> component as a card.', () => {
    const component = shallow(<MegaCard {...defaultProps} type="card" />)
    expect(component).toMatchSnapshot()
  })

  test('Should render the <MegaCard /> component as a pill.', () => {
    const component = shallow(<MegaCard {...defaultProps} type="pill" />)
    expect(component).toMatchSnapshot()
  })

  test('Should render the <MegaCard /> component as default.', () => {
    const component = shallow(<MegaCard {...defaultProps} />)
    expect(component).toMatchSnapshot()
  })
})
