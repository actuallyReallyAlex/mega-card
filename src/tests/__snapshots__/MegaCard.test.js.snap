// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MegaCard /> Should render the <MegaCard /> component as a card. 1`] = `ShallowWrapper {}`;

exports[`<MegaCard /> Should render the <MegaCard /> component as a pill. 1`] = `ShallowWrapper {}`;

exports[`<MegaCard /> Should render the <MegaCard /> component as default. 1`] = `ShallowWrapper {}`;
